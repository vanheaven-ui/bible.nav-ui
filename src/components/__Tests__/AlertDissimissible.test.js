import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import '@testing-library/jest-dom';
import AlertDisimissible from '../AlertDissimissible';

describe('AlertDissimissible component', () => {
  const handleClick = () => '';

  it('renders without crashing', () => {
    ReactDOM.render(<AlertDisimissible handleClick={handleClick} error="" />, rootDiv());
  });

  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <AlertDisimissible handleClick={handleClick} error="" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct wrapper element', () => {
    const { container } = render(
      <AlertDisimissible handleClick={handleClick} error="" />,
    );
    expect(container.firstChild.classList.contains('show')).toBe(true);
  });
});
