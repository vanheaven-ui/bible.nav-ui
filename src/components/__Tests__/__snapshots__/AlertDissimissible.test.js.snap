// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDissimissible component does not change unexpectedly 1`] = `
<p
  className="show"
>
  
  <button
    onClick={[Function]}
    type="button"
  >
    X
  </button>
</p>
`;
